src/components/lib/sl/sl.c

```diff
@@ -343,20 +411,36 @@ sl_thd_alloc(cos_thd_fn_t fn, void *data)
 	struct sl_thd *t;
 
 	sl_cs_enter();
-	t = sl_thd_alloc_intern(fn, data, NULL, 0);
+	t = sl_thd_alloc_intern(fn, data);
 	sl_cs_exit();
 
 	return t;
 }
 
-/* Allocate a thread that executes in the specified component */
 struct sl_thd *
-sl_thd_comp_alloc(struct cos_defcompinfo *comp)
+sl_thd_aep_alloc(cos_aepthd_fn_t fn, void *data, int owntc)
 {
 	struct sl_thd *t;
 
 	sl_cs_enter();
-	t = sl_thd_alloc_intern(NULL, NULL, comp, 1);
+	t = sl_thd_aep_alloc_intern(fn, data, NULL, owntc ? SL_FLAG_OWNTC: 0);
+	sl_cs_exit();
+
+	return t;
+}
+
+/* sl object for inithd in the child comp */
+struct sl_thd *
+sl_thd_comp_init(struct cos_defcompinfo *comp, int owntc)
```

Comment:
As you commented, do we care about the case where the child uses the same tcap?  Perhaps so.  Maybe worth a 5 minute discussion on Wed.  Can keep it as it is (perhaps with a new name) for now.